plugin/src/main/java/org/opennms/timeseries/cortex/ResultMapper.java

```diff
@@ -95,4 +111,48 @@ static Metric appendExternalTagsToMetric(final Metric metric, final KeyValueStor
             return builder.build();
         } else return metric;
     }
+
+    private static List<Metric> parseMetrics(String json)  {
+        try (JsonParser p = JSON_FACTORY.createParser(json)) {
+
+            if (p.nextToken() != JsonToken.START_OBJECT) {
+                throw new IOException("Invalid JSON");
+            }
+
+
+            while (p.nextToken() != JsonToken.END_OBJECT) {
+                if ("data".equals(p.getCurrentName()) && p.nextToken() == JsonToken.START_ARRAY) {
+
+                    if (p.nextToken() != JsonToken.START_OBJECT) {
+                        return Collections.emptyList();
+                    }
+
+                    MappingIterator<Map<String, String>> iterator =
+                            MAPPER.readValues(p, new TypeReference<Map<String, String>>() {});
+
+                    List<Metric> list = new ArrayList<>();
+                    while (iterator.hasNext()) {
+                        list.add(toMetricFromMap(iterator.next()));
+                    }
+                    return list;
+                }
+                p.skipChildren();
+            }
+        } catch (JsonParseException e) {
+            throw new RuntimeException(e);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return Collections.emptyList();
+    }
+
+    private static List<Metric> enrichWithExternalTags(
+            List<Metric> metrics,
+            KeyValueStore store) {
+        return metrics.parallelStream()
+                .map(metric -> appendExternalTagsToMetric(metric, store))
```

Comment:
Note that for each call, we are invoking keyValueStore from Postgres, this would create too many connections on Postgres. 

We probably need to optimize    appendExternalTagsToMetric where it can fetch all externalTagsRaw once instead of per key.  That would probably be better.

Reply:
remove `parallelStream` for now. We can have another issue that will do batch fetching for external tags.